Lib/asyncio/unix_events.py

```diff
@@ -488,7 +493,37 @@ def __repr__(self):
             info.append('closed')
         return '<{}>'.format(' '.join(info))
 
-    def _read_ready(self):
+    def _readinto_buffer_ready(self):
+        try:
+            buf = self._protocol.get_buffer(-1)
+            if not len(buf):
+                raise RuntimeError('get_buffer() returned an empty buffer')
+        except (SystemExit, KeyboardInterrupt):
+            raise
+        except BaseException as exc:
```

Comment:
Is there some specific need to handle BaseException here instead of Exception?

Reply:
👍